sbin/submitty_daemon_jobs/submitty_jobs/generate_pdf_images.py

```diff
@@ -13,28 +13,55 @@ def __init__(self, page_number: int, coordinates: Sequence[float]):
         self.coordinates = coordinates
 
 
-def main(pdf_file_path: str, redactions: List[Redaction]):
+def main(pdf_file_path: str, output_dir: str, redactions: List[Redaction]):
     directory = os.path.dirname(pdf_file_path)
     if directory:
         os.chdir(os.path.dirname(pdf_file_path))
+    # Ensure the output directory exists
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+    # Delete any existing images that match the pattern
+    for filename in os.listdir(output_dir):
+        if filename.startswith(os.path.basename(pdf_file_path)[:-4]) and filename.endswith(".jpg"):
+            os.remove(os.path.join(output_dir, filename))
     try:
         pdfPages = PdfReader(pdf_file_path, strict=False)
-        with open(pdf_file_path, 'rb') as open_file:
+        with open(pdf_file_path, "rb") as open_file:
             imagePages = convert_from_bytes(
                 open_file.read(),
-                )
+            )
         for page_number in range(len(pdfPages.pages)):
-            image_filename = pdf_file_path[:-4] + '_' + str(page_number + 1).zfill(3) + '.jpg'
-            imagePages[page_number].save(image_filename,
-                                         "JPEG", quality=20, optimize=True)
+            image_filename = (
+                output_dir
+                + "/."
+                + os.path.basename(pdf_file_path)[:-4]
+                + "_page_"
+                + str(page_number + 1).zfill(2)
+                + ".jpg"
+            )
+            imagePages[page_number].save(
+                image_filename, "JPEG", quality=20, optimize=True
+            )
+            img = Image.open(image_filename)
+            draw = ImageDraw.Draw(img)
             for redaction in redactions:
-                if redaction.page_number != page_number:
+                if redaction.page_number != page_number + 1:
                     continue
-                img = Image.open(image_filename)
-                draw = ImageDraw.Draw(img)
-                draw.rectangle(redaction.coordinates, fill="black")
-                img.save(image_filename,
-                         "JPEG", quality=20, optimize=True)
+                square_size = 25
+
+                x0 = int(redaction.coordinates[0] * img.size[0])
+                y0 = int(redaction.coordinates[1] * img.size[1])
+                x1 = int(redaction.coordinates[2] * img.size[0])
+                y1 = int(redaction.coordinates[3] * img.size[1])
+
+                for y in range(y0, y1, square_size):
+                    for x in range(x0, x1, square_size):
+                        fill_color = "black" if ((x // square_size + y // square_size) % 2 == 0) else "grey"
+                        draw.rectangle(
+                            [x, y, x + square_size, y + square_size], fill=fill_color
+                        )
```

Comment:
What's going on here?  Why do we need these loops?  Comments could help resolve this confusion...